vendor/models/offer.py

```diff
@@ -94,7 +100,27 @@ def description(self):
             return self.products.all().first().description
     
     def savings(self, currency=DEFAULT_CURRENCY):
-        savings = self.get_msrp(currency) - self.current_price()
+        """
+        Gets the savings between the difference between the product's msrp and the currenct price
+        """
+        savings = self.get_msrp(currency) - self.current_price(currency)
         if savings < 0:
             return Decimal(0).quantize(Decimal('.00'), rounding=ROUND_UP)
         return Decimal(savings).quantize(Decimal('.00'), rounding=ROUND_UP)
+
+    def get_best_currency(self, currency=DEFAULT_CURRENCY):
+        """
+        Gets best currency for prodcuts available in this offer
+        """
+        supported_currencies = []
+        for product in self.products.all():
+            for supported_currency in product.meta['msrp'].keys():
+                if supported_currency != 'default':
+                    supported_currencies.append(supported_currency)
+        
+        if currency in supported_currencies:
+            return currency
+        
+        # TODO: Should this instead throw a warning or an error?
+        return self.products.all().first().meta['msrp']['default']
```

Comment:
Look at this for getting the currency instead of appending to supported_currencies:

https://www.w3schools.com/python/ref_set_intersection.asp

It will make it easier to narrow down the list to only include what exists across all the products in the offer.

Generate a list of sets based on the currency keys that are not "default".  Then with intersect, you can check the full list.  Will have to pop the first or last item off first to use as the base set.

Reply:
added the intersection method to cross check available currencies